# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/l10n/Localization.cpp
# Commit: 85194e5b7a9e
# Full Hash: 85194e5b7a9e6ed9114ee46aef0b5efb819cca9b
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-04 03:56:33
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 181487bdf6c0 -r 85194e5b7a9e intl/l10n/Localization.cpp
--- a/intl/l10n/Localization.cpp	Tue Aug 03 16:25:13 2021 +0000
+++ b/intl/l10n/Localization.cpp	Tue Aug 03 16:25:13 2021 +0000
@@ -237,31 +237,21 @@
 
 void Localization::OnChange() { ffi::localization_on_change(mRaw.get()); }
 
-void Localization::AddResourceId(const nsAString& aResourceId) {
-  NS_ConvertUTF16toUTF8 resId(aResourceId);
-  ffi::localization_add_res_id(mRaw.get(), &resId);
+void Localization::AddResourceId(const nsACString& aResourceId) {
+  ffi::localization_add_res_id(mRaw.get(), &aResourceId);
 }
 
-uint32_t Localization::RemoveResourceId(const nsAString& aResourceId) {
-  NS_ConvertUTF16toUTF8 resId(aResourceId);
-  return ffi::localization_remove_res_id(mRaw.get(), &resId);
+uint32_t Localization::RemoveResourceId(const nsACString& aResourceId) {
+  return ffi::localization_remove_res_id(mRaw.get(), &aResourceId);
 }
 
-void Localization::AddResourceIds(const nsTArray<nsString>& aResourceIds) {
-  nsTArray<nsCString> resIds(aResourceIds.Length());
-  for (const auto& resId : aResourceIds) {
-    resIds.AppendElement(NS_ConvertUTF16toUTF8(resId));
-  }
-  ffi::localization_add_res_ids(mRaw.get(), &resIds);
+void Localization::AddResourceIds(const nsTArray<nsCString>& aResourceIds) {
+  ffi::localization_add_res_ids(mRaw.get(), &aResourceIds);
 }
 
 uint32_t Localization::RemoveResourceIds(
-    const nsTArray<nsString>& aResourceIds) {
-  nsTArray<nsCString> resIds(aResourceIds.Length());
-  for (const auto& resId : aResourceIds) {
-    resIds.AppendElement(NS_ConvertUTF16toUTF8(resId));
-  }
-  return ffi::localization_remove_res_ids(mRaw.get(), &resIds);
+    const nsTArray<nsCString>& aResourceIds) {
+  return ffi::localization_remove_res_ids(mRaw.get(), &aResourceIds);
 }
 
 already_AddRefed<Promise> Localization::FormatValue(